examples/gradients.rs

```diff
@@ -1,33 +1,35 @@
-use std::time::Instant;
-
-use femtovg::{renderer::OpenGl, Align, Baseline, Canvas, Color, Paint, Path, Renderer};
-use glutin::{
+use femtovg::{renderer::OpenGl, Canvas, Color, Paint, Path, Renderer};
+use instant::Instant;
+use resource::resource;
+use winit::{
     event::{Event, WindowEvent},
     event_loop::{ControlFlow, EventLoop},
-    window::WindowBuilder,
-    ContextBuilder,
 };
 
+mod helpers;
+use helpers::PerfGraph;
+
 fn main() {
-    let window_size = glutin::dpi::PhysicalSize::new(1000, 670);
-    let el = EventLoop::new();
-    let wb = WindowBuilder::new()
-        .with_inner_size(window_size)
-        .with_resizable(false)
-        .with_title("Gradient test");
+    #[cfg(not(target_arch = "wasm32"))]
+    helpers::start(1000, 670, "Gradient test", false);
+    #[cfg(target_arch = "wasm32")]
+    helpers::start();
+}
 
-    let windowed_context = ContextBuilder::new().build_windowed(wb, &el).unwrap();
-    let windowed_context = unsafe { windowed_context.make_current().unwrap() };
+#[cfg(not(target_arch = "wasm32"))]
+use glutin::PossiblyCurrent;
 
-    let renderer = OpenGl::new_from_glutin_context(&windowed_context).expect("Cannot create renderer");
-    let mut canvas = Canvas::new(renderer).expect("Cannot create canvas");
-    canvas.set_size(
-        window_size.width as u32,
-        window_size.height as u32,
-        windowed_context.window().scale_factor() as f32,
-    );
+#[cfg(target_arch = "wasm32")]
+use winit::window::Window;
+
+fn run(
+    mut canvas: Canvas<OpenGl>,
+    el: EventLoop<()>,
+    #[cfg(not(target_arch = "wasm32"))] windowed_context: glutin::WindowedContext<PossiblyCurrent>,
+    #[cfg(target_arch = "wasm32")] window: Window,
```

Comment:
The "#ifdeffery" here is quite painful, but I also see how this isn't easily addressed. The only way that I could imagine is to wrap event loop, context and (optional) Window in another type and let the example provide a callback that is called from `run()` and which (unconditionally) takes the `&WindowedContext`, `&Window` and `&Event`.